modules/lang-painless/src/main/java/org/opensearch/painless/action/PainlessExecuteAction.java

```diff
@@ -577,6 +579,13 @@ static Response innerShardOperation(Request request, ScriptService scriptService
                     double result = scoreScript.execute(null);
                     return new Response(result);
                 }, indexService);
+            } else if (scriptContext == UpdateScript.CONTEXT) {
```

Comment:
We don't need the `UpdateScript.CONTEXT` right?

Reply:
We don't need it to fix the bug. I added it to reproduce the issue in a Unit Test.